od.proto

```diff
@@ -40,15 +40,39 @@ message OD {
 // TODO Do we want to model multi-modal trips, like driving to a park-and-ride
 // and taking transit? Or using an e-scooter for last mile?
 enum Mode {
-  // Single-occupancy vehicle? Rideshare? Passengers vs drivers?
-  Driving = 0;
-  Cycling = 1;
-  Walking = 2;
-  // Distinguish buses; ferries, trains?
-  PublicTransit = 3;
-  Carpool = 4;
-  ElectricMicromobility = 5;
-  // More?
+  // Human powered modes
+  WALK = 0;
+  JOG = 1;
+  PUSH_SCOOTER = 2;
+  PEDAL_CYCLE = 3;
+
+  // Electric mobility and 2 wheelers
+  MICROMOBILITY = 4;
+  ELECTRIC_SCOOTER = 5;
+  ELECTRIC_CYCLE = 6;
+  MOBILITY_SCOOTER = 7;
+  MOTORCYCLE = 8;
+  // From STATS19
+  // Motorcycle - unknown cc 125cc and under 50cc
+  // over 125cc up to 500cc Motorcycle over 500cc
+
+  // Public transport
+  PUBLIC_TRANSPORT = 9;
+  MINIBUS = 10;
+  BUS = 11;
+  LIGHT_RAIL = 12;
+  RAIL = 13;
+
+  // Cars/Vans
+  CAR = 14;
```

Comment:
Same here -- what's the difference between `CAR`, `CAR_DRIVER`, and `CAR_PASSENGER`, and these other categories?

We could try to get more structured with the use of a car -- something like:

```
message CarTransport {
  VehicleType vehicle_type = 1;
  OccupantType occupant_type = 2;
}

enum VehicleType {
  UNKNOWN = 1,
  MICRO = 2,
  SMALL = 3,
  LARGE = 4,
  VAN = 5,
  TRUCK = 6,
}

enum OccupantType {
  UNKNOWN = 1,   // Maybe some datasets don't distinguish drivers from passengers  
  DRIVER = 1,
  PASSENGER_IN_FAMILY_VEHICLE = 2,
  PASSENGER_IN_CARPOOL = 3,
  PASSENGER_IN_RIDESHARE = 4,
}
```

And maybe more attributes about the vehicle are important -- gas, diesel, hybrid, electric?

I don't know if having more detail here is useful or not. 

1) Are there any known use cases for these distinctions now?
2) Do any of the data sources cover this nuance?

Reply:
>     1. Are there any known use cases for these distinctions now?

Yes: we have data on car size from MOT data and it would be great to show that. Longer term it would be ideal to have continuous variables associated with each car.

> 
>     2. Do any of the data sources cover this nuance?

Yes, MOT data and UK vehicle fleet data.

I like the idea of dividing it up into sub-categories but that has some negatives: it's hard to understand and for users to use, more complex and feels a bit like mission creep. Having lots of bins where people can put things is fine I think and I doubt having a few separate bins will make a big difference.

Reply:
Maybe we can compromise on the Cartesian product of vehicle types and occupant types, with a few less cases in each category.

Vehicles: van, truck, micro car, small car, large car, unknown
Occupant type: driver, passenger (let's throw away the distinction of calling a private rideshare like Lyft and carpooling with a friend)

So we'd have 7 * 2 = 14 categories. `VAN_DRIVER`, `VAN_PASSENGER`, `MICRO_CAR_DRIVER`, `MICRO_CAR_PASSENGER`, etc.

Similar to the comment above, it's unambiguous which category something should map to. Each of these sets is mutually exclusive.

Reply:
Sounds reasonable. So `VEHICLE_UNKNOWN` and `CAR_UNKNOWN` or just the former? This is progress!